bin/add_vcf_header_info_records.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python3
+
+"""
+Add INFO fields to VCF Header
+"""
+
+import argparse
+import os
+from typing import List, Dict, Optional
+
+VALID_NUMBER_SPECIAL_CHARS = ("A", "R", "G", ".")
+VALID_INFO_TYPES = ("Integer", "Float", "Flag", "Character", "String")
+
+
+def main() -> None:
+    """
+    Add new INFO rows to VCF, output to new VCF file.
+    """
+    args = parse_arguments()
+
+    if not os.path.exists(args.vcf):
+        raise FileNotFoundError(f"Input VCF file '{args.vcf}' does not exist.")
+
+    processed_header_data = process_info_args(args.info)
+    info_lines_to_be_added = generate_info_lines(processed_header_data)
+    add_info_to_vcf(args.vcf, args.output, info_lines_to_be_added)
+
+
+def parse_arguments() -> argparse.Namespace:
+    """
+    Parse command-line arguments.
+
+    Returns:
+        argparse.Namespace: Parsed arguments.
+    """
+    parser = argparse.ArgumentParser(description="Add INFO fields to a VCF header.")
+
+    parser.add_argument(
+        "--vcf",
+        required=True,
+        help="Path to the input VCF file.",
+    )
+
+    parser.add_argument(
+        "--info",
+        action="append",
+        nargs=6,  # Expect exactly six arguments
+        metavar=("ID", "Number", "Type", "Description", "Source", "Version"),
+        required=True,
+        help=(
+            "Define an INFO field in the header. Provide exactly six arguments: "
+            "ID, Number, Type, Description, Source, and Version. Use an empty string "
+            "(e.g., '') for Source and Version if not applicable. This argument can "
+            "be repeated to add multiple INFO fields."
+        ),
+    )
+    parser.add_argument(
+        "--output",
+        required=True,
+        help="Path to the output VCF file.",
+    )
+
+    return parser.parse_args()
+
+
+def process_info_args(info_args: List[List[str]]) -> List[Dict[str, Optional[str]]]:
+    """
+    Process INFO arguments, ensuring valid Source and Version usage.
+
+    Args:
+        info_args (List[List[str]]): Parsed INFO arguments.
+
+    Returns:
+        List[Dict]: INFO fields as dictionaries.
+    """
+
+    id_: str
+    number: str
+    type_: str
+    description: str
+    source: Optional[str]
+    version: Optional[str]
+
+    processed_info = []
+    for info in info_args:
+        id_, number, type_, description, source, version = info
+
+        if not source:
+            source = None
+
+        if not version:
+            version = None
+
+        validate_info_fields(number=number, info_type=type_)
+
+        processed_info.append(
+            {
+                "ID": id_,
+                "Number": number,
+                "Type": type_,
+                "Description": description,
+                "Source": source,
+                "Version": version,
+            }
+        )
+    return processed_info
+
+
+def validate_info_fields(*, number: str, info_type: str) -> None:
```

Comment:
Googling around a bit, the "*" is to enforce key-based parameter passing or?

Why not just skip the "*" and still pass by keys?

Reply:
> I think I saw a risk that the position of the existing two args would change, e.g. if it would turn out that the ID needed to be validated in some way, then that arg would naturally be inserted before the other two in the function definition.

Hmm. For me I would be with content with letting the future coder doing this making sure that the arguments ends up in the right place. I think the asterisk confuses more than it helps. But this is not an opinion I will hold strongly here.

> Another solution would be to implement that info field class you suggested above and pass that instead.

👍